### PR TITLE
feat: build Information dialog component to show confirmation or alert

### PR DESCRIPTION
**Pull request by yzhao583**
Jira: [https://issues.redhat.com/browse/FLPATH-853](https://issues.redhat.com/browse/FLPATH-853)

Build a information dialog component to ask user to confirm or show alert to user.

image
image

### CODE REVIEW

**Review by jkilzi — CHANGES_REQUESTED** [review]
Looking good!
IMO there is still some room for small improvements. I left some comments in the review on that regard.

File: plugins/orchestrator/src/components/InfoDialog.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    const DialogActions = () => (
+      <>
+        <Button onClick={handleClose} color="primary">
+          Disagree
+        </Button>
+        <Button onClick={handleSubmit} color="primary" autoFocus>
+          Agree
+        </Button>
+      </>
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.stories.tsx:88]: sure

File: plugins/orchestrator/src/components/InfoDialog.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+const ConfirmationDialogContent = () => (
+  <div>
+    Are you sure you want to submit? By click the submit button, you cannot
+    change the action
+  </div>
+);
+const AlertDialogContent = () => (
+  <div>
+    Let Google help apps determine location. This means sending anonymous
+    location data to Google, even when no apps are running.
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.stories.tsx:28]: This is the text copied from an example, but sure.

File: plugins/orchestrator/src/components/InfoDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  IconButton,
+  makeStyles,
+  Typography,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+
+export type InfoDialogProps = {
+  title: string;
+  open: boolean;
+  close?: () => void;
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.tsx:18]: ok

File: plugins/orchestrator/src/components/InfoDialog.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        <Button onClick={handleSubmit} color="primary" autoFocus>
+          Agree
+        </Button>
+      </>
+    );
+
+    return (
+      <InfoDialog
+        {...args}
+        close={handleClose}
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.stories.tsx:94]: done

File: plugins/orchestrator/src/components/InfoDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+  return (
+    <Dialog onClose={_ => close} open={open} ref={forwardedRef}>
+      <DialogTitle>
+        <Box>
+          <Typography variant="h5">{title}</Typography>
+          <IconButton
+            className={classes.closeBtn}
+            aria-label="close"
+            onClick={close}
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.tsx:48]: done

File: plugins/orchestrator/src/components/InfoDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    right: 8,
+    top: 8,
+  },
+}));
+
+export const RefForwardingInfoDialog: ForwardRefRenderFunction<
+  ParentComponentRef,
+  InfoDialogProps
+> = (props, forwardedRef): JSX.Element | null => {
+  const { title, open, close, children, dialogActions } = props;
```
[the remark this answers is not in the document]
Comment (yzhao583) [reply plugins/orchestrator/src/components/InfoDialog.tsx:37]: done

**Comment by yzhao583** [comment]
@jkilzi Thanks for the review! Just modified the PR based on feedbacks, please review again, thanks!

**Comment by jkilzi** [comment]
@yzhao583 PTAL at [this](https://github.com/janus-idp/backstage-plugins/actions/runs/7820291139/job/21334646923?pr=1176#step:3:410).

File: plugins/orchestrator/package.json [hunk 7]
```diff
@@ -80,7 +80,8 @@
     "react-use": "^17.4.0",
     "swr": "^2.0.0",
     "uuid": "^9.0.1",
-    "vscode-languageserver-types": "^3.16.0"
+    "vscode-languageserver-types": "^3.16.0",
+    "@storybook/preview-api": "^7.5.3"
```
Comment (jkilzi) [inline plugins/orchestrator/package.json:84]: Please move it to `devDependency`

**Review by jkilzi — CHANGES_REQUESTED** [review]

**Review by jkilzi — APPROVED** [review]

**Pull request merged**